internal/services/machinelearning/machine_learning_workspace_hub_resource.go

```diff
@@ -0,0 +1,504 @@
+	Location                    string                                     `tfschema:"location"`
+	ResourceGroupName           string                                     `tfschema:"resource_group_name"`
+	KeyVaultID                  string                                     `tfschema:"key_vault_id"`
+	StorageAccountID            string                                     `tfschema:"storage_account_id"`
+	ContainerRegistryID         string                                     `tfschema:"container_registry_id"`
+	ApplicationInsightsID       string                                     `tfschema:"application_insights_id"`
+	PublicNetworkAccess         string                                     `tfschema:"public_network_access"`
+	WorkspaceId                 string                                     `tfschema:"workspace_id"`
+	FriendlyName                string                                     `tfschema:"friendly_name"`
+	Description                 string                                     `tfschema:"description"`
+	PrimaryUserAssignedIdentity string                                     `tfschema:"primary_user_assigned_identity"`
+	Identity                    []identity.ModelSystemAssignedUserAssigned `tfschema:"identity"`
+	Encryption                  []encryptionModel                          `tfschema:"encryption"`
+	Tags                        map[string]string                          `tfschema:"tags"`
+}
+
+type encryptionModel struct {
+	KeyVaultID             string `tfschema:"key_vault_id"`
+	KeyID                  string `tfschema:"key_id"`
+	UserAssignedIdentityID string `tfschema:"user_assigned_identity_id"`
+}
+
+type MachineLearningWorkspaceHubResource struct{}
+
+var _ sdk.Resource = MachineLearningWorkspaceHubResource{}
+
+func (r MachineLearningWorkspaceHubResource) ResourceType() string {
+	return "azurerm_machine_learning_workspace_hub"
+}
+
+func (r MachineLearningWorkspaceHubResource) ModelObject() interface{} {
+	return &machineLearningWorkspaceHubModel{}
+}
+
+func (r MachineLearningWorkspaceHubResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return workspaces.ValidateWorkspaceID
+}
+
+func (r MachineLearningWorkspaceHubResource) Arguments() map[string]*pluginsdk.Schema {
+	arguments := map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"location": commonschema.Location(),
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"description": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+
+		"key_vault_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateKeyVaultID,
+			// TODO -- remove when issue https://github.com/Azure/azure-rest-api-specs/issues/8323 is addressed
+			DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
can you please reach out to the service team about this? the linked issue is 4 years old and there is no indication of any progress on the service team fixing it. could we get an update here?

Reply:
Will check with the service team, the issue appears to been fxied.